File: vpbuf/examples/uno/rel1/nodejs/src/stack.js
#!/usr/bin/nodejs
// Software released under the MIT license (see project root for license file)

"use strict";

const persist = require("./persist");
var vp = require('./vp_uno');

// ---------------------------------------------------------------------

(function() {
    var library = {
        build_uno_deck: function(max_count) {
            var d = new persist.Deck()
            var build_arr = []
            var count = 0

            for (const c of persist.color_arr) {

                for (var k = 0; k < 2; k++) {
                    build_arr.push(new persist.Skip(++count, c))
                    build_arr.push(new persist.Reverse(++count, c))
                    build_arr.push(new persist.DrawTwo(++count, c))
                }

                build_arr.push(new persist.Value(++count, 0, c))

                for (var k = 1; k < 10; k++) {
                    build_arr.push(new persist.Value(++count, k, c))
                    build_arr.push(new persist.Value(++count, k, c))
                }
            }

            d.cards = build_arr.slice(count - max_count)

            return d
        }
    };

    // -----------------------------------------------------------------
    // begin command line processing

    const args = process.argv.slice(2)

    if (args.length == 0) {
        console.log('usage is: stack <deck_name> [ <card_count> ]')
    } else {
        var deck_name = args[0]
        var count = 3

        if (args.length > 1)
            count = args[1]

        console.log('[vrange='+ vp.get_low_version() +'-'+ vp.get_high_version()
                            +', nodejs] STACK '+ count + ' cards')

        var deck = library.build_uno_deck(count)

        persist.save_deck(deck_name, deck)
    }
})();

// ---------------------------------------------------------------------

File: vpbuf/examples/uno/rel1/python/stack.py
#!/usr/bin/python3

# Software released under the MIT license (see project root for license file)

from pyamf import amf3
from vp_uno.persist import *
from vp_uno.vp_uno import *
from vp_uno.tools import *

import argparse

# ------------------------------------------------------------------------------

def build_poker_deck(card_count):
    deck = Deck()
    build_arr = []

    count = 0
    for c in color_arr:

        for k in range(0,2):
            build_arr.append(Skip(++count, c))
            build_arr.append(Reverse(++count, c))
            build_arr.append(DrawTwo(++count, c))

        build_arr.append(Value(++count, 0, c))

        for k in range(1,10):
            build_arr.append(Value(++count, k, c))
            build_arr.append(Value(++count, k, c))

    deck.cards = build_arr[len(build_arr) - card_count:]

    return deck

# ------------------------------------------------------------------------------

def main():
    # process args

    p = argparse.ArgumentParser()
    p.add_argument("deck_name")
    p.add_argument("count", nargs='?', default=3)

    args = p.parse_args()

    print('[vrange=',get_low_version(),'-',get_high_version(),
            ', python] STACK ', args.count, ' cards', sep='')

    deck = build_poker_deck(int(args.count))

    save_deck(args.deck_name, deck)

if __name__ == '__main__':
    main()

# ------------------------------------------------------------------------------